SimCalorimetry/HGCalSimProducers/interface/HGCDigitizerBase.h

```diff
@@ -10,157 +10,153 @@
 #include "DataFormats/HGCDigi/interface/HGCDigiCollections.h"
 #include "FWCore/Utilities/interface/EDMException.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
-#include "CLHEP/Random/RandGaussQ.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
 
-typedef float HGCSimEn_t;
-typedef std::array<HGCSimEn_t,6> HGCSimHitData;
-typedef std::unordered_map<uint32_t, HGCSimHitData> HGCSimHitDataAccumulator;
+#include "SimCalorimetry/HGCalSimProducers/interface/HGCFEElectronics.h"
+
+typedef float HGCSimData_t;
+
+//15 time samples: 9 pre-samples, 1 in-time, 5 post-samples
+typedef std::array<HGCSimData_t,15> HGCSimHitData;
+
+//1st array=energy, 2nd array=energy weighted time-of-flight
+typedef std::unordered_map<uint32_t, std::array<HGCSimHitData,2> > HGCSimHitDataAccumulator; 
 
 template <class D>
```

Comment:
(minor) one-letter types are not so good for an implementation with so many lines